lightning/src/chain/channelmonitor.rs

```diff
@@ -386,6 +394,10 @@ enum OnchainEvent {
 	MaturingOutput {
 		descriptor: SpendableOutputDescriptor,
 	},
+	FundingSpendConfirmation {
```

Comment:
"Funding output has been either by us or remote, when it's mature we return the balance availability to the caller". Though I wonder if we couldn't just add a field to `MaturingOutput`, they're both generated in `transactions_confirmed` ?

Reply:
Hmm? `MaturingOutput` is not created in `transactions_confirmed`, and it could be skipped if we have no to-self output or on a cooperative close transaction.

Reply:
Well it's created in `is_paying_spendable_output` which is called by `transactions_confirmed`. Though yes no guarantee it's  generated for cooperative closing transactions, I wasn't sure if you intended to cover this case.

Should user also expect to get the txid/on_local_output on _revoked_ commitment transaction ?

Reply:
Yes, we should cover cooperative closing transactions so that we stop providing a balance after a channel is closed. I imagine users will basically ask for the current set of spendable balances and add it to the GUI-displayed balance so we should be careful to not expose too much balance and have it persist.

Indeed, tracking balances for revoked spend needs to happen, but is left largely implemented in this PR (I believe we'll always just say no balance to spend. I'd updated the docs to mention this a few days ago after you'd started review initially.